packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -675,6 +748,29 @@ class Scaffold extends StatefulWidget {
     return scaffoldScope.geometryNotifier;
   }
 
+  /// Sets the [ScaffoldGeometry.floatingActionButtonNotchMaker] for the closest
+  /// [Scaffold] ancestor of the given context if one exist.
+  ///
+  /// It is guaranteed that `notchMaker` will only be used for making notches
+  /// in the top edge of the [BottomAppBar], the start and end offsets given to
+  /// it will always be on the top edge of the [BottomAppBar], the start offset
+  /// will be to the left of the floating action button's bounds, and the end
+  /// offset will be to the right of the floating action button's bounds.
+  ///
+  /// Returns null if there was no [Scaffold] ancestor.
+  /// Otherwise, returns a [VoidCallback] that clears the notch maker that was
+  /// set.
+  ///
+  /// Callers must invoke the callback when the notch is no longer required.
+  /// This method is typically called from [State.didChangeDependencies] and the
+  /// callback should then be invoked from [State.deactivate].
```

Comment:
This should probably mention that only the last call will be remembered, so if there's two FloatingActionButtons, only one will be represented in the ScaffoldGeometry.

Reply:
Added. Though it feels a bit over-verbose to mention that a setX method overrides the previous value of X...